ansible_collections/arista/avd/roles/eos_designs/python_modules/mlag/__init__.py

```diff
@@ -0,0 +1,361 @@
+        """
+        if self._mlag is True:
+            return super().render()
+        return {}
+
+    @cached_property
+    def _mlag(self):
+        return get(self._hostvars, "switch.mlag")
+
+    @cached_property
+    def _mlag_l3(self):
+        return get(self._hostvars, "switch.mlag_l3")
+
+    @cached_property
+    def _mlag_peer_l3_vlan(self):
+        return get(self._hostvars, "switch.mlag_peer_l3_vlan")
+
+    @cached_property
+    def _mlag_peer_vlan(self):
+        return get(self._hostvars, "switch.mlag_peer_vlan", required=True)
+
+    @cached_property
+    def _trunk_groups_mlag_name(self):
+        return get(self._hostvars, "switch.trunk_groups.mlag.name", required=True)
+
+    @cached_property
+    def _trunk_groups_mlag_l3_name(self):
+        return get(self._hostvars, "switch.trunk_groups.mlag_l3.name", required=True)
+
+    @cached_property
+    def _underlay_routing_protocol(self):
+        return get(self._hostvars, "switch.underlay_routing_protocol", required=True)
+
+    @cached_property
+    def _p2p_uplinks_mtu(self):
+        return get(self._hostvars, "p2p_uplinks_mtu", required=True)
+
+    @cached_property
+    def _mlag_port_channel_id(self) -> int:
+        return int(get(self._hostvars, "switch.mlag_port_channel_id", required=True))
+
+    @cached_property
+    def _mlag_peer(self):
+        return get(self._hostvars, "switch.mlag_peer", required=True)
+
+    @cached_property
+    def _mlag_ibgp_origin_incomplete(self):
+        return get(self._hostvars, "switch.mlag_ibgp_origin_incomplete")
+
+    @cached_property
+    def _underlay_rfc5549(self):
+        return get(self._hostvars, "underlay_rfc5549")
+
+    @cached_property
+    def _bgp_as(self):
+        return get(self._hostvars, "switch.bgp_as", required=True)
+
+    @cached_property
+    def _mlag_peer_ip(self):
+        return get(self._hostvars, "switch.mlag_peer_ip", required=True)
+
+    @cached_property
+    def spanning_tree(self):
+        if self._mlag_peer_l3_vlan is not None:
+            vlans = unique([self._mlag_peer_vlan, self._mlag_peer_l3_vlan])
+            return {"no_spanning_tree_vlan": list_compress(vlans)}
+
+        return {"no_spanning_tree_vlan": self._mlag_peer_vlan}
+
+    @cached_property
+    def vlans(self):
+        vlans = {}
+        if self._mlag_peer_l3_vlan is not None:
+            vlans.update(
+                {
+                    self._mlag_peer_l3_vlan: {
+                        "tenant": "system",
+                        "name": "LEAF_PEER_L3",
+                        "trunk_groups": [self._trunk_groups_mlag_l3_name],
+                    }
+                }
+            )
+        vlans.update(
+            {
+                self._mlag_peer_vlan: {
+                    "tenant": "system",
+                    "name": "MLAG_PEER",
+                    "trunk_groups": [self._trunk_groups_mlag_name],
+                }
+            }
+        )
```

Comment:
Given that we are updating only one key, I find this following syntax more readable (missing black formating as I am free typing this)

```suggestion
            vlans[self._mlag_peer_l3_vlan] = {
                        "tenant": "system",
                        "name": "LEAF_PEER_L3",
                        "trunk_groups": [self._trunk_groups_mlag_l3_name],
                    }

        vlans.update[self._mlag_peer_vlan] =  {
                    "tenant": "system",
                    "name": "MLAG_PEER",
                    "trunk_groups": [self._trunk_groups_mlag_name],
                }
```

Reply:
Fixed